src/interdiff.c

```diff
@@ -98,15 +118,74 @@ static int ignore_components = 0;
 static int ignore_components_specified = 0;
 static int unzip = 0;
 static int no_revert_omitted = 0;
+static int use_colors = 0;
+static int color_option_specified = 0;
 static int debug = 0;
 
 static struct patlist *pat_drop_context = NULL;
 
 static struct file_list *files_done = NULL;
+
 static struct file_list *files_in_patch2 = NULL;
 static struct file_list *files_in_patch1 = NULL;
 
-/* checks whether file needs processing and sets context */
+/*
+ * Print colored output using a variadic format string.
+ * This avoids the need for memory allocation and deallocation
+ * that was required with the colorize_line approach.
+ */
+ static void
+ print_color (FILE *output_file, enum line_type type, const char *format, ...)
+ {
+	 const char *color_start = "";
+	 const char *color_end = "";
+	 va_list args;
+
+	 /* Only colorize if colors are enabled AND we're outputting to stdout */
+	 if (use_colors && output_file == stdout) {
+		 switch (type) {
+		 case LINE_CONTEXT:
+			 color_start = COLOR_CONTEXT;
+			 break;
+		 case LINE_ADDED:
+			 color_start = COLOR_ADDED;
+			 break;
+		 case LINE_REMOVED:
+			 color_start = COLOR_REMOVED;
+			 break;
+		 case LINE_HEADER:
+			 color_start = COLOR_HEADER;
+			 break;
+		 case LINE_FILE:
+			 color_start = COLOR_FILE;
+			 break;
+		 case LINE_HUNK:
+			 color_start = COLOR_HUNK;
+			 break;
+		 }
+
+		 if (color_start[0] != '\0') {
+			 color_end = COLOR_RESET;
+		 }
+	 }
+
+	 /* Print color start code */
+	 if (color_start[0] != '\0') {
```

Comment:
I noticed other parts of the code largely don't use braces for if-statements that have a single line in the body, perhaps these braces should be removed along with the other occurrences?